src/main/kotlin/org/rust/cargo/util/TasksUtil.kt

```diff
@@ -0,0 +1,59 @@
+package org.rust.cargo.util
+
+import com.google.common.util.concurrent.*
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.progress.impl.CoreProgressManager
+import com.intellij.util.concurrency.AppExecutorUtil
+
+object TasksUtil
+
+/**
+ * Schedules task to run asynchronously (on the application thread-pool)
+ *
+ * NOTA BENE: This would occupy 2 (sic!) threads: one carrying payload, and the second one to piggyback results of the
+ *            former right into `ListenableFuture`. We need to address that
+ */
+fun Task.Backgroundable.enqueue(): ListenableFuture<*> =
```

Comment:
Good catch. Though i (personally) think some IDEA's API is a pure crap, the overhead for this facade is too big.